Lib/glyphsLib/types.py

```diff
@@ -405,3 +413,58 @@ def plistValue(self, format_version=2):
         # TODO write hex bytes in chunks and split over multiple lines
         # for better readability, like the fonttools xmlWriter does
         return "<%s>" % binascii.hexlify(self).decode()
+
+
+class IndexPath(ValueType):
+    """A list of indexes.
+
+    It is analogous to `NSIndexPath`, which is a list of indices that together
+    represent the path to a specific location in a tree of nested arrays. This
+    class is used internally by Glyphs for storing properties of hints,
+    including `origin`, `other1`, `other2`, and `target`.
+
+    The most common case is that it is a list of two integers pointing at a real
+    node. However, it can also be three or four integers. Moreover, in the case
+    of `origin` and `target`, it can be a list containing a single string.
+    """
+
+    def __init__(
+        self,
+        value: Union[int, str, List[Union[int, str]]],
+        value2: Optional[int] = None,
+        value3: Optional[int] = None,
+        value4: Optional[int] = None,
+    ):
+        if value4 is not None:
+            self.value = [value, value2, value3, value4]
+        elif value3 is not None:
+            self.value = [value, value2, value3]
+        elif value2 is not None:
+            self.value = [value, value2]
+        elif isinstance(value, (list, tuple)):
+            self.value = value
+        else:
+            self.value = self.fromString(value)
+
+    def fromString(self, string: str) -> List[Union[int, str]]:
```

Comment:
@schriftgestalt, I have discovered one interesting case, which seems to contradict the Markdown file. It says that `origin` and `target` are `tuple`'s, even when they contain a single string. So it should be `target = (up)`, `target (down)`, etc. However, this is not what I see in Glyphs files. Unfortunately, this is not covered in `GlyphsFileFormatv3.glyphs`. Now I am not sure what is right. It seems that `origin` and `target` are not tuples when a string is the only element, in which case it would be `target = up`, `target = down`, etc. If this is the case, it would be nice to clarify in the Markdown document.